### PR TITLE
[Score mode] Fixed score not given to the correct winner

### PR DESCRIPTION
**Pull request by eaocto**

### CODE REVIEW

**Comment by DO97** [comment]
@eaocto @dvlwj When a round ends with a draw card, the next player will not draw the amount of cards to draw (because the round ends, this is acceptable) and this amount will be drawn the next time (next round) a draw card is played.

Another bug is related to the total points; when a game ends, the player points will not be resetted to 0, because the points will be equal to the previous game ones

**Comment by eaocto** [comment]
@DO97 can you elaborate more related to total points problem? Maybe giving a sample case and a step by step break down if possible. From my understanding by looking at the code the "game" ends when there's only a single player left; the "game" thus is destroyed by "game manager". After that when starting a new "game" "game manager" will create new instance of game, therefore the score should be zero.

As for the "draw card at the round end"; which case is true below?
a.) +2 card > round ends > new round > first player draw 2 card
b.) +2 card > round ends > new round > ... normal play ... > player can't play card in hand and then draw 2 card
c.) +2 card > round ends > new round > ... normal play ... > +2 card > next player draw 4 card
d.) +2 card > round ends > new round > ... normal play ... > +2 card > next player draw 2 card

**Comment by DO97** [comment]
@eaocto The problem regards the new game. A score mode game ends correctly, player A wins (ex. 234 points) and player B loses (ex. 89 points). But if they (player A and player B) start a new game (score mode), their points will still be 234 and 89.

A) When a player draws a card, he'll draw the total amount from the previous round (ex. if the previous round has ended with a +2 and another, the player will draw 4 cards instead of one).
C) Yes, similar to A. If a player plays a draw card, the next player (if he has no draw cards) will draw 2 cards + the draw counter from the previous round.

Basically, the counter of cards to draw is not resetted to 0 when the round ends.

**Comment by DO97** [comment]
@eaocto
